### PR TITLE
added canary dependencies

### PR DESCRIPTION
**Pull request by pe-vsn**
It is really weird to me to that git is comparing these binaries and whatnot. I hope I am misunderstanding how this is supposed to work.

@dw-vsn

### CODE REVIEW

**Comment by pe-vsn** [comment]
I think you want to be ignoring tar.xz files, right? It is not currently in the .gitignore though.

**Comment by pe-vsn** [comment]
I added a flag to gpm, also made a pull request there. It will no longer install the packages we get. But it still handles commit hashes like before. Hopefully this is a good solution.

**Comment by danw-volley** [comment]
That sounds good, thanks.

File: tools/gpm [hunk 1]
```diff
@@ -74,13 +75,26 @@ set_dependencies() {
   done < <(echo "$deps")
   wait
 
-  while read package version; do
-    echo ">> Building package "$package""
-    go install "$package"
-  done < <(echo "$deps")
+  if [[ $no_install != true ]]
```
Comment (danw-volley) [inline tools/gpm:80]: OK, I see you are just following existing formatting in this file. That makes sense since we didn't write gpm to start with.

**Comment by danw-volley** [comment]
+1

File: tools/gpm (hunk 1 above)
Comment (pe-vsn) [inline tools/gpm:80]: Sure, I was just following the style guide they set forth since this is not our script.

**Comment by pe-vsn** [comment]
This last commit was just making sure my updated gpm script still passes the tests (`make test`) in the original library so I am not breaking this if we want to use it differently in the future.

**Comment by pe-vsn** [comment]
This is still not up-to-date with the pull request I have with the gpm team. But it seems like that might take a while. I think we can merge this for now so we can keep moving forward and then pull upstream later when my PR is accepted. How does that sound?

**Comment by danw-volley** [comment]
Yes Phil, I agree we push forward and not worry about upstream gpm.

**Comment by pe-vsn** [comment]
Great, I'll be awaiting your plus one on the last small updates up there then.

**Comment by danw-volley** [comment]
+1

**Pull request merged**